db/migrations/20220608090500_remove_duplicated_tenant_ids_org_ids.go

```diff
@@ -0,0 +1,263 @@
+				// code is that if an ID has already been processed, it will be skipped to avoid problems.
+				for orgId, ids := range orgIdIds {
+					logging.Log.Infof(`[org_id: %s][ids: %v] Processing duplicate tenant group`, orgId, ids)
+
+					// Grab the first tenant ID as the main one.
+					if len(ids) > 0 {
+						logging.Log.Infof(`[main_tenant_id: %d] Selecting main tenant id`, ids[0])
+						mainTenantId = ids[0]
+					}
+
+					// duplicateCounter will help us keep track of the duplicate count for a particular
+					// "orgId" value.
+					var duplicateCounter uint = 1
+
+					// Since we've already grabbed the first ID from the slice, just take the rest starting from there.
+					for _, id := range ids[1:] {
+						logging.Log.Infof(`[tenant_id: %d][org_id: %s] Processing tenant`, id, orgId)
+
+						// Update the old tenant ID from all the tables.
+						for _, table := range updatableTables {
+							err := updateTenantFromTable(tx, table, id, mainTenantId)
+							if err != nil {
+								return err
+							}
+						}
+
+						// Mark the tenant as duplicate.
+						mark := fmt.Sprintf(duplicateTenantMark, orgId, duplicateCounter)
+						err := markOrgIdAsDuplicate(tx, mark, id)
+						if err != nil {
+							return err
+						}
+
+						duplicateCounter++
+					}
+				}
+
+				return nil
+			})
+
+			return err
+		},
+		// There is nothing we can rollback in this migration. Once we have updated all the tenant IDs from the
+		// subresources we would not know how to "split" again those tenant IDs into the duplicated tenants.
+		Rollback: func(db *gorm.DB) error {
+			return nil
+		},
+	}
+}
+
+// updateTenantFromTable updates the "tenant_id" column to the specified value from the given table.
+func updateTenantFromTable(tx *gorm.DB, table string, oldTenantId int64, newTenantId int64) error {
+	updateTableSql := `
+		UPDATE
+			"%s"
+		SET
+			"tenant_id" = ?
+		WHERE
+			"tenant_id" = ?
+	`
+
+	updateQuery := fmt.Sprintf(updateTableSql, table)
+
+	err := tx.
+		Debug().
+		Exec(updateQuery, newTenantId, oldTenantId).
+		Error
+
+	if err != nil {
+		return err
+	}
+
+	logging.Log.Infof(`Updated table "%s". Set tenant "%d" to "%d"`, table, oldTenantId, newTenantId)
+
+	return nil
+}
+
+// markExternalTenantAsDuplicate marks the old duplicated tenant as duplicate. It modifies the "external_tenant" column
+// to add a value that will let us know that these were duplicated tenants, and that they shouldn't have any related
+// subresources.
+func markExternalTenantAsDuplicate(tx *gorm.DB, mark string, tenantId int64) error {
+	err := tx.
+		Debug().
+		Model(&model.Tenant{}).
+		Where("id = ?", tenantId).
+		Update("external_tenant", mark).
+		Error
+
+	if err != nil {
+		return err
+	}
+
+	logging.Log.Infof(`Tenant "%d"'s "external_tenant" marked as processed duplicate with mark "%s"`, tenantId, mark)
+
+	return nil
+}
+
+// markExternalTenantAsDuplicate marks the old duplicated tenant as duplicate. It modifies the "org_id" column to add a
```

Comment:
Or the name of the function is - I see the `org_id` reference later on.

Reply:
Yes, initially I was doing one pass and updating both columns to the same value, but then I thought that doing a second pass with org ids wouldn't hurt, since the redundant updates would essentially be NOPs. That's why we have two functions for the migration. I'll update the docs.